src/main/java/net/replaceitem/integratedcircuit/mixin/BlockRenderViewMixin.java

```diff
@@ -0,0 +1,10 @@
+package net.replaceitem.integratedcircuit.mixin;
+
+import net.minecraft.world.BlockRenderView;
+import net.minecraft.world.RedstoneView;
+import org.spongepowered.asm.mixin.Mixin;
+
+@Mixin(BlockRenderView.class)
+public interface BlockRenderViewMixin extends RedstoneView {
```

Comment:
Maybe it is? Ehm, actually it definitely is. I thought the interface you're casting to had to extend the one you're casting from, but [apparently not](https://stackoverflow.com/questions/13219097/casting-between-interfaces). Neat

Reply:
I just mean the fact that it's an empty mixin class that does nothing

Reply:
Actually, no, this is necessary, it crashes without it. This is there because I want to access `RedstoneView` methods from a `BlockRenderView` for the color provider in IntegratedCircuitClient, and the actual class that gets passed in doesn't implement RedstoneView, so still we need it

Reply:
> I just mean the fact that it's an empty mixin class that does nothing

Is this the wrong way to make BlockRenderView extend RedstoneView? Sorry I'm not too good with mixins, but it seems to work..

Reply:
Oh yeah true in that case it actually does something. I assumed BlockRenderView already extended RedstoneView in vanilla. Seems hacky though. Alternatively we could do something like `((ChunkRendererRegion) view).world` which should always be a client world, which is a RedstoneView. We would probably need a mixin to get `world` though. Not sure what's the better way here.

Or we change outputStrength to not only be output, but more like a `portStrength`, so with input ports it gets set to whats coming in from outside (server side) and the client renderer can just always use that value for the tinting?

Reply:
> Or we change outputStrength to not only be output, but more like a portStrength, so with input ports it gets set to whats coming in from outside (server side) and the client renderer can just always use that value for the tinting?

I think this makes the most sense to me